staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt-config.go

```diff
@@ -0,0 +1,304 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flowcontrol
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	fq "k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing"
+	kubeinformers "k8s.io/client-go/informers"
+	cache "k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	rmtypesv1a1 "k8s.io/api/flowcontrol/v1alpha1"
+)
+
+// initializeConfigController sets up the controller that processes
+// config API objects.
+func (reqMgmt *requestManagementSystem) initializeConfigController(informerFactory kubeinformers.SharedInformerFactory) {
+	reqMgmt.configQueue = workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(200*time.Millisecond, 8*time.Hour), "req_mgmt_config_queue")
+	fci := informerFactory.Flowcontrol().V1alpha1()
+	pli := fci.PriorityLevelConfigurations()
+	fsi := fci.FlowSchemas()
+	reqMgmt.plInformer = pli.Informer()
+	reqMgmt.plLister = pli.Lister()
+	reqMgmt.fsInformer = fsi.Informer()
```

Comment:
> no need to keep the informer instance, conventionally

same here

Reply:
Is this a documented convention?  I do not think there is only one right way to do it.

We need to check sync in Run.  Perhaps you are thinking of what the sample-controller does, it saves the HasSynced function rather than the informer.  I see no harm in saving the informer rather than its HasSynced.

In some other controllers, the Indexer of the informer is needed by the queue workers.  It would be awkward to save multiple attributes of the informer rather than saving just the informer.

Reply:
> We need to check sync in Run. Perhaps you are thinking of what the sample-controller does

we can keep the sync func as field then, e.g. `podHasSynced cache.InformerSynced`


> In some other controllers, the Indexer of the informer is needed by the queue workers.

commonly, direct access to the indexer in the informer is not recommended, unless we customized index-func of the informer. we're supposed to call listers instead

Reply:
Yes, I was thinking of controllers that use custom indices.  My point is that some controllers need to save more than the sync function.  So it is not wrong to save the informer. In the case at hand we have a choice of saving either the sync function or the informer; either would work, neither is wrong.